### PR TITLE
Make sure that we have valid values for setting the ContentSize of the ScrollView on iOS

### LINKED ISSUE #15079: [regression/7.0.86] On iOS grid rows that are set to star display as auto when view is added later that spans all rows and columns
### Description

After updating to 7.0.86 we are seeing an issue when adding a menu overlay to page grids.

The page overlay is effectively a menu system we are converting from Xamarin Forms where we detect the view the user has long tapped on and display an opaque view on top with a menu of options the user can perform.

The page grid is a simple enough layout, an auto row 0 (a toolbar), a star row 1 (for data), and auto row 2 (quick navigation options). Within row 1 we display another grid with 5 or 6 star rows, and each of those contains a scrollview. This allows the user to scroll data within each row if there is more information to display.

When adding the page overlay to the page grid we span all its columns and rows. Worked perfectly previously, but now all the rows within the page expand and display all the data within the scrollview as if they were auto.

### Steps to Reproduce

This will be a pain to create a repo, but Im happy to share video, screenshots, or even a TestFlight build to demo the issue.
I *could* at a push invite someone to the app repo in Azure DevOps but I will want to speak to someone first.

### Link to public reproduction project repository

Nope

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS all

### Did you find any workaround?

Previous Maui versions worked, so the latest grid fixes appear to have broken something else.
EDIT:
Adding `<MauiVersion>7.0.59</MauiVersion>` to the project and the grid behaves correctly.
7.0.86 fixes a lot of other layout issues btw.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change

Updated version of https://github.com/dotnet/maui/pull/14143. Simplified things a little, added comments for clarity.

fixes #15079
fixes (but not really) #14257

### CODE REVIEW

**Review by rmarinho — APPROVED** [review]

**Comment by JohnHDev** [comment]
@hartez any way of tracking which build this fix is going/gone in to? We use gittag's in Jira for this, any equivalent here?

**Comment by hartez** [comment]
We have a bot that labels PRs with their initial release versions; I'm not sure why it hasn't labeled this. But it's available in the latest .NET 8 preview; it has not yet been backported to 7. The backport/suggested label means it will be in our next backport review; assuming it meets the criteria we'll backport it and it will show up in the following service release.

**Comment by JohnHDev** [comment]
@hartez thank you for the update, that's helpful.
I have to ask though, why aren't bug fixes going directly into .NET 7 as that is the current production version and then merged forward into .NET 8?

**Comment by hartez** [comment]
Because all code changes to the stable version run a non-zero risk of regression, and not all bugs are severe enough to warrant that risk. And some (very few) bug fixes require breaking changes, so they can't be done in 7 at all. So we do the bug fixes in 8, where we still have time to surface problems prior to stable release. For bugs where the severity is high enough (weighed against the complexity of the fix), we create a backport PR and add it to a service release for 7 (and possibly 6, if it meets the bar we've set).

**Comment by softlion** [comment]
There are so few people using Maui net7 in production, why not include breaking changes.

**Comment by DDHSchmidt** [comment]
> There are so few people using Maui net7 in production, why not include breaking changes.

Following that logic will lead to a self-fulfilling prophecy:
"Why is no one using .Net 7?"
"There are too many bugs in it"
"So why aren't they fixing them?"
"Everyone fled to .Net 8, so why bother fixing 7 at all?"

For us, the last 2 Service Releases after 7.0.59 contained so many breaking regression bugs that we had to sit them out, which means no updates for almost 4 months.
While e.g. the touted memory leak fixes would be very much welcomed, they're useless [if all of our grids break](https://github.com/dotnet/maui/issues/14520), [again](https://github.com/dotnet/maui/issues/15018) and [again](https://github.com/dotnet/maui/issues/15079) :(

We're eagerly awaiting every new .Net 7 SR, hoping that we can finally upgrade but at this point it really feels like Microsoft wants everyone to be a .Net 8 beta tester anyway...

**Comment by JohnHDev** [comment]
> Because all code changes to the stable version run a non-zero risk of regression, and not all bugs are severe enough to warrant that risk. And some (very few) bug fixes require breaking changes, so they can't be done in 7 at all. So we do the bug fixes in 8, where we still have time to surface problems prior to stable release. For bugs where the severity is high enough (weighed against the complexity of the fix), we create a backport PR and add it to a service release for 7 (and possibly 6, if it meets the bar we've set).

@hartez thanks, do you know if this will be merged back into .NET 7? If not, when is .NET 8 due for release?
We have a hard deadline of May 1st, 2024 to move off Xamarin. .NET 7 is current production and we cannot move to .NET 8 until it is released and we can't start the migration to .NET Maui until our features can be migrated.

**Comment by softlion** [comment]
> Following that logic will lead to a self-fulfilling prophecy:
> ...
> it really feels like Microsoft wants everyone to be a .Net 8 beta tester anyway...

I'm not sure if you are with or against me at that point.

I do have the same requirements. 

Both a production app using net7 that is waiting for an update that I can't release because of "suddenly" poor performances everywhere (not only in collection views).I don't care about breaking changes as people using net7 are on the edge anyway, and can bear with breaking changes.

And a large xamarin app that I would like to start migrating at least 6 months before the end of xamarin support.

**Comment by JohnHDev** [comment]
@hartez I haven't seen a reply to my previous question but I see the .NET 8 release date is November, why are production bugs not being fixed in .NET 7 first? Do we really have to wait 5 months for fixes that MS doesn't deem as High?
You marked by ticket as fixed and closed, but I have no idea if the fix will be in .NET 7 or not. All I know is that it might be fixed (I haven't tested it) in future version which may or may not have even more regressions.

A November release date only gives us a 6 month window to migrate our monolithic Xamarin Forms apps, when we don't know if it will even work. Please please please, you (MS) have got to give us better support than this.

@davidortinau are you aware of the process described above?

**Comment by hartez** [comment]
> do you know if this will be merged back into .NET 7? If not, when is .NET 8 due for release?

It's been marked for consideration as a backport (that's what the "backport suggested" label is for); we'll be evaluating it soon and determining whether we can backport it. 

.NET 8 should be hitting this autumn. In the meantime, the preview versions are being released so you can try them out and see what migration issues you run into.

**Comment by JohnHDev** [comment]
@hartez Do you have a label for regression, as this worked previously in .NET 7. I would think regressions are always backported, but I don't want to assume anything when it comes to .NET Maui.

I would also suggest you reopen my bug report until it has been merged into .NET 7, as that is the version that the bug was found.

**Comment by JohnHDev** [comment]
Just watched https://www.youtube.com/watch?v=arxsDF__uFA.

So not only is @davidortinau aware of this process of fixing in .NET 8 (a version in beta) and then considering merging back into .NET 7 (you know, the production version of your code that everyone uses) he mentioned it is the official MS process.

So as of writing, it could be 5-6 months before we see some production fixes. I clearly disagree with this approach, MS is ignoring its customers issues and fixing them in a version *we cannot use*. No, we are not going to waste our time 'trying out a beta version of .NET'.

Long story short, we are no longer considering migrating to .NET Maui at this time but we will take another look in November with the release of .NET 8. I manage a team of 9 Xamarin developers, I cannot in good conscience allocate anyone to look at .NET Maui in its current state.

Please consider extending Xamarin Forms support for a further 6 to 12 months.

**Comment by Inrego** [comment]
Also want to add that this is a regression. Grids were working fine in .NET 7. Then I updated Visual Studio and now everwhere the app looks like shit. I've spent hours upon hours trying to come up with workarounds to no avail. Being that it's so slow to build and deploy debug builds for iOS (#14560) it's really tedious.
